include/boost/math/special_functions/jacobi_theta.hpp

```diff
@@ -0,0 +1,646 @@
+
+// Policied versions - parameterized by tau
+template <class RealType, class Policy>
+inline RealType jacobi_theta1tau(RealType z, RealType tau, const Policy& pol);
+template <class RealType, class Policy>
+inline RealType jacobi_theta2tau(RealType z, RealType tau, const Policy& pol);
+template <class RealType, class Policy>
+inline RealType jacobi_theta3tau(RealType z, RealType tau, const Policy& pol);
+template <class RealType, class Policy>
+inline RealType jacobi_theta4tau(RealType z, RealType tau, const Policy& pol);
+
+// Policied m1 functions
+template <class RealType, class Policy>
+inline RealType jacobi_theta3m1(RealType z, RealType q, const Policy& pol);
+template <class RealType, class Policy>
+inline RealType jacobi_theta4m1(RealType z, RealType q, const Policy& pol);
+template <class RealType, class Policy>
+inline RealType jacobi_theta3m1tau(RealType z, RealType tau, const Policy& pol);
+template <class RealType, class Policy>
+inline RealType jacobi_theta4m1tau(RealType z, RealType tau, const Policy& pol);
+
+#ifdef __JACOBI_THETA_USE_IMAGINARY
+// The following _IMAGINARY theta functions assume imaginary z and are for
+// internal use only. The z argument is scaled by tau, and sines and cosines
+// are replaced with hyperbolic sines and cosines to accommodate the imaginary
+// argument. (Recall sinh=(exp(x)-exp(-x))/2 and cosh=(exp(x)+exp(-x))/2)
+//
+// The return values are scaled by exp(-tau*z²/π)/sqrt(tau).
+//
+// Using the convention τ'=-1/τ and noting that both τ and τ' are always
+// imaginary (at least in our applications), the _IMAGINARY functions are used
+// to implement the following four relations:
+//
+// [20.7.30] sqrt(-iτ)θ₁(z|τ) = -i*exp(iτ'z²/π)*θ₁(zτ'|τ')
+// [20.7.31] sqrt(-iτ)θ₂(z|τ) =    exp(iτ'z²/π)*θ₄(zτ'|τ')
+// [20.7.32] sqrt(-iτ)θ₃(z|τ) =    exp(iτ'z²/π)*θ₃(zτ'|τ')
+// [20.7.33] sqrt(-iτ)θ₄(z|τ) =    exp(iτ'z²/π)*θ₂(zτ'|τ')
+template <class RealType, class Policy>
+inline RealType
+_IMAGINARY_jacobi_theta1tau(RealType z, RealType tau, const Policy& pol) {
+    BOOST_MATH_STD_USING
+    unsigned n = 0;
+    RealType eps = policies::get_epsilon<RealType, Policy>();
+    RealType /* q_n, */ delta, delta1, delta2, result = RealType(0);
+
+    do {
+        delta1 = exp(tau*(z*(RealType(2*n+1) - z/constants::pi<RealType>())
+                    -constants::pi<RealType>() * RealType(n + 0.5)*RealType(n + 0.5)));
+        // q_n = exp(-tau * constants::pi<RealType>() * RealType(n + 0.5)*RealType(n + 0.5) );
+        delta2 = exp(tau*(z*(-RealType(2*n+1) - z/constants::pi<RealType>())
+                    -constants::pi<RealType>() * RealType(n + 0.5)*RealType(n + 0.5)));
+
+        if (n%2) {
+            delta = delta2 - delta1;
+        } else {
+            delta = delta1 - delta2;
+        }
+
+        result += delta;
+        n++;
+    } while (abs(delta * sqrt(tau)) > eps || (abs(result * sqrt(tau)) > eps * eps && abs(delta/result) > eps));
```

Comment:
Could you add a comment about the error analysis that leads to this termination criteria?

Reply:
The `sqrt(tau)` stuff was removed in the latest commit, but I can clarify what is left.